### PR TITLE
Issue #2645 - Change behavior of Response.ok

### PR DESCRIPTION
**Pull request by AlexPHorta**
As discussed on issue #2645, I added a success attribute to
requests.codes. Now, Response.ok checks if the response's status code
is listed on requests.codes.success.

### CODE REVIEW

File: requests/status_codes.py [hunk 1]
```diff
@@ -89,3 +89,18 @@
...
+    203,
+    204,
+    205,
+    206,
+    207,
+    208,
+    226,
+}
+
+setattr(codes, 'success', _success)
```
Comment (sigmavirus24) [inline requests/status_codes.py]: I think this is over-engineering the solution. This also isn't the appropriate object to tack "success" onto when it's not a status code and the attribute you're setting is not an integer (like everything else on codes has been historically).

I'm okay having something like

``` py
# requests/status_codes.py
success_set = {
    # ...
}
```

``` py
# requests/models.py
from . import status_codes

class Response(..):
    @property
    def ok(self):
        return self.status_code in status_codes.success_set
```

Comment (Lukasa) [inline requests/status_codes.py]: :+1:

Comment (sigmavirus24) [inline requests/status_codes.py]: Oh, also note that on Py26, you can't use set literal notation. You have to do

``` py
success_set = set([
    200,
    201,
    # ...
])
```

**Comment by AlexPHorta** [comment]
I get it, it's just that I thought the agreed solution was to add the success attribute to requests.codes. I'll update the commit right now.

**Comment by Lukasa** [comment]
This LGTM. We can't merge until we're ready to do 3.0.0, though.

**Comment by kennethreitz** [comment]
An additional check should be added for entire 2xx range after the registered codes are checked (or perhaps the '-1 index +1 :' slice). Both for future-proofing and a server might be using an esoteric number honoring the numbering scheme.

**Comment by kennethreitz** [comment]
Also, I've shared some thoughts about this on https://github.com/kennethreitz/requests/issues/2645.

Basically, no behavioral modifications to `ok`. A new attribute can be added for the 200-only functionality, if this is still desired. 

I will need to think for a bit about the naming. feel free to use `is_successful` as a placeholder for now.

---

Maintainers: **do not merge** without my approval.

**Comment by kennethreitz** [comment]
P.S. everybody gets cake :sparkles: :cake: ✨

**Pull request closed without merging**